### PR TITLE
Combo Distributions

### PR DESCRIPTION
**Pull request by embedthis**
Combination (Amalgamated) distributions are single source file distributions that catentate all the required source files. To do this, symbols in the various source files must not clash with other source files.

This patch has the changes to the library/*.c to enable catenation.

### CODE REVIEW

**Comment by sbutcher-arm** [comment]
This PR, similar to it's sister PR #287, is asking the project to support a new type of distribution. If we adopt it, it needs to be properly maintained, to avoid being broken in the future, which is otherwise likely to happen again and again. And that means integrating this into our testing and our CI, potentially with a script to check for duplicated symbols.

At the moment, we have PR's coming that resolve the naming problems with `mbedtls_zeroize()` and also the duplication of file I/O functions, in the #421 PR. So some of these issues are planned to be resolved.

This doesn't fix all the other duplicates variable names. However, I don't think `#undef` is the best way to address them, and I would prefer to rename variables where possible.

I don't think we want to obstruct using the library in this way, but I don't think this PR is the way we'd like to do it. If anyone out there (including @ghost) still cares, please raise an issue concerning the duplicate variable names.

If they want to submit code, I would suggest PR's that rename variables, rather than the `#undef`. There's no need to do that for the file I/O functions or `mbedtls_zeroize()` as we have fixes for that in the pipeline. And also a script to check for duplicate symbol names is essential to maintain this.

In conclusion - I'm closing this PR and rejecting it. However - if someone wants to reopen the PR and resubmit it as I describe above, we'd welcome the PR.
